modules/msg/option_inc/default_cfe_msg_hdr_pri.h

```diff
@@ -0,0 +1,94 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/**
+ * @file
+ *
+ * Define cFS standard full header
+ *  - Avoid direct access for portability, use APIs
+ *  - Used to construct message structures
+ */
+
+#ifndef DEFAULT_CFE_MSG_HDR_PRI_H
+#define DEFAULT_CFE_MSG_HDR_PRI_H
+
+/*
+ * Include Files
+ */
+
+#include "common_types.h"
+#include "ccsds_hdr.h"
+#include "cfe_msg_sechdr.h"
+#include "cfe_msg_api_typedefs.h"
+
+/*
+ * Type Definitions
+ */
+
+/**********************************************************************
+ * Structure definitions for full header
+ *
+ * These are based on historically supported definitions and not
+ * an open source standard.
+ */
+
+/**
+ * \brief Full CCSDS header
+ */
+typedef struct
+{
+    CCSDS_PrimaryHeader_t Pri; /**< \brief CCSDS Primary Header */
+} CCSDS_SpacePacket_t;
+
+/**
+ * \brief cFS generic base message
+ *
+ * This provides the definition of CFE_MSG_Message_t
+ */
+union CFE_MSG_Message
+{
+    CCSDS_SpacePacket_t CCSDS;                             /**< \brief CCSDS Header (Pri or Pri + Ext) */
+    uint8               Byte[sizeof(CCSDS_SpacePacket_t)]; /**< \brief Byte level access */
+};
```

Comment:
I'm assuming this structure is analogous to CFE_SB_Message_t from older builds of cFS. SB used to enforce a minimum 32-bit alignment for CFE_SB_Msg_t instances by including a uint32 member in the union. This does not appear to be preserved in CFE_MSG_Message_t. Is this intentional? Are there any negative implications for not forcing a 32-bit alignment?

Reply:
Intentional and avoids unintended padding or restrictions when using the `CFE_MSG_TelemetryHeader_t` or `CFE_MSG_CommandHeader_t` when defining message structures.  The `CFE_SB_Buffer_t` forces alignment:
https://github.com/nasa/cFE/blob/c8b5e00bb722d8f59787ae8f82634feba5d39f60/modules/core_api/fsw/inc/cfe_sb_api_typedefs.h#L125-L131

This makes it so `CFE_SB_Buffer_t` (what you get from `CFE_SB_RecieveBuffer`) can safely be casted to any custom message structure.  When sending a message structure with `CFE_SB_TransmitMsg` the message is copied into a buffer.